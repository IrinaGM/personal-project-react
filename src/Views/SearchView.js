import React, { useState } from "react";
import Button from 'react-bootstrap/Button';
import {BASE_URL} from "../Components/Consts"


const SearchView = (props) => {
    const [userName,setUserName] = useState('');
    const [errMsg,setErrMsg] = useState('');

    const hangdleSearch = () => {
        fetch(`${BASE_URL}/${userName}`)
            .then(response => response.json())
            .then(user => user.message ? setErrMsg(user.message) : props.toggleSearchView(user.login))
            .catch(error => setErrMsg(error.message)); 
    };

  return <div>
        <h1> GitHub Stats Search </h1>
        <input className="mb-2 mt-2 w-50"  id="userNameInput" type="text" onChange={event => setUserName(event.target.value)}></input><br/>
        <Button variant="primary" onClick={hangdleSearch}>Search</Button>
        {errMsg ? <p> User {errMsg} </p> : ''}
    </div>
};

export default SearchView;