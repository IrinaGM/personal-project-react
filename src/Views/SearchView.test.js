import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';

import SearchView from './SearchView';


describe('SearchView',() => {
    test('Should render properly', () => {
        const renderer = new ShallowRenderer();
        const element = renderer.render(<SearchView />);
        expect(element).toMatchSnapshot();
    });
});
