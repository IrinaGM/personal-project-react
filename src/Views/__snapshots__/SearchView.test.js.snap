// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchView Should render properly 1`] = `
<div>
  <h1>
     GitHub Stats Search 
  </h1>
  <input
    className="mb-2 mt-2 w-50"
    id="userNameInput"
    onChange={[Function]}
    onKeyPress={[Function]}
    type="text"
  />
  <br />
  <ForwardRef
    active={false}
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Search
  </ForwardRef>
  
</div>
`;
